test/ArbitrumFoundationVestingWallet.t.sol

```diff
@@ -0,0 +1,194 @@
+        ) = deployAndInit();
+        assertEq(foundationVestingWallet.beneficiary(), beneficiary, "beneficiary set");
+
+        vm.prank(vestingWalletOwner);
+        foundationVestingWallet.setBeneficiary(newBeneficary);
+        assertEq(foundationVestingWallet.beneficiary(), newBeneficary, "new beneficiary set");
+    }
+
+    function testOwnlyOwnerCanSetBeneficiary() external {
+        (
+            ArbitrumFoundationVestingWallet foundationVestingWallet,
+            L2ArbitrumToken token,
+            L2ArbitrumGovernor gov
+        ) = deployAndInit();
+
+        vm.prank(rando);
+        vm.expectRevert("Ownable: caller is not the owner");
+        foundationVestingWallet.setBeneficiary(address(newBeneficary));
+    }
+
+    function testRelease() external {
+        (
+            ArbitrumFoundationVestingWallet foundationVestingWallet,
+            L2ArbitrumToken token,
+            L2ArbitrumGovernor gov
+        ) = deployAndInit();
+        assertEq(token.balanceOf(beneficiary), 0, "beneficiary has no tokens");
+
+        // sanity: ensure test env is att a < starttime timestamp
+        assertTrue(block.timestamp < startTime, "pre start time");
+
+        vm.startPrank(beneficiary);
+        foundationVestingWallet.release();
+        assertEq(token.balanceOf(beneficiary), 0, "beneficiary still has no tokens");
+
+        vm.warp(1500);
+
+        foundationVestingWallet.release(address(token));
+        assertEq(token.balanceOf(beneficiary), 500, "beneficiary got tokens");
+        vm.stopPrank();
+
+        vm.prank(vestingWalletOwner);
+        foundationVestingWallet.setBeneficiary(newBeneficary);
+
+        vm.warp(1600);
+
+        vm.prank(newBeneficary);
+        foundationVestingWallet.release(address(token));
+
+        assertEq(token.balanceOf(newBeneficary), 100, "new beneficiary got tokens");
+
+        assertEq(token.balanceOf(beneficiary), 500, "prev beneficiary didn't get more tokens");
+    }
+
+    function testOnlyBeneficiaryCanRelease() public {
+        (
+            ArbitrumFoundationVestingWallet foundationVestingWallet,
+            L2ArbitrumToken token,
+            L2ArbitrumGovernor gov
+        ) = deployAndInit();
+        vm.prank(rando);
+        vm.expectRevert("ArbitrumFoundationVestingWallet: not beneficiary");
+        foundationVestingWallet.release(address(token));
+    }
+
+    function testOnlyOwnerCanMigrate() public {
+        (
+            ArbitrumFoundationVestingWallet foundationVestingWallet,
+            L2ArbitrumToken token,
+            L2ArbitrumGovernor gov
+        ) = deployAndInit();
+        vm.startPrank(rando);
+        vm.expectRevert("Ownable: caller is not the owner");
+        foundationVestingWallet.migrateTokensToNewWallet(address(token), rando);
+
+        vm.expectRevert("Ownable: caller is not the owner");
+        foundationVestingWallet.migrateEthToNewWallet(rando);
+    }
+
+    function testMigrateTokensToNewWallet() public {
```

Comment:
could we also have a test case that explicitly tests `testMigrateTokensToNewWalletWithSlowerVesting` and `testMigrateTokensToNewWalletWithFasterVesting`
serves almost as documentation to the intended path on how to configure those intended paths